### DISCUSSION TITLE
[REQ] JS or other Scripting Engine

### DISCUSSION THREAD

**Discussion by 00xZEROx00** [mozilla/hubs#3347] [Spoke Questions] [answered]
**Is your feature request related to a problem? Please describe.**
I want to know if there is a plan for add interactivity in hubs, like js or other web language, or even in visual programming.
I think the main issue is to maintain the semplicity of Hubs / Spoke, or worse, the security of Hubs (maybe an evil scripting...)
But lack of a minimum of scripting / visual programming, is a big missing feature for me...
Like i want to make a button to open a door, start an animation on a model, play an audio file.

**Describe the solution you'd like**
I've see a nice concepts like Rec Room or Neos VR where users can visual programming some things, like mini circuits, and or logic operators, and so on...
So basically we need to add some simple behaviours to objects, the best bet for me is pre-made the interactivity in Spoke, so end-users won't complicate their life with code or programming, but only content creators can if they want, add interactivity to the scene, i'm not a dev, so i don't know if this implicit some security questions, or other issues.
Maybe we can add this as an additional branch with an scripting engine, based on web technologies like JS...
Unreal Engine blueprint system is a thing, maybe implement a system like that in Spoke should be awesome, but for now not overwhelming with functions, just some actions like, press this open this, press this play this, or move this, or walk thru this point change a material...

**Describe alternatives you've considered**
An alternative is to not to rush into interactivity at the moment, and made Spoke and Hubs more mature, even if at actual state is really good. Then think to add interactivity in another moment...

Thank you for your reading, i don't know if this is already asked, but hope i receive a clear response, have a good day! <3

**Comment by robinkwilson** [comment, marked as answer]
This would be great as an issue (or a comment on an existing issue)! https://github.com/mozilla/hubs/issues

This is highly asked for, please search for "extensibility" in the hubs repo to find an existing issue.
